n2y/converter.py

```diff
@@ -389,41 +390,53 @@ def __init__(self, client: Client, block, get_children=True):
         self.file = File(block['image'])
 
     def to_pandoc(self):
-        url = None
-        if self.file.type == "external":
-            url = self.file.url
-        elif self.file.type == "file":
-            url = self.file.download()
         caption = RichTextArray(self.caption)
+        external = self.file.type == "external"
+        file = self.file.type == "file"
+
+        if external:
+            url = self.file.create_unique_filename(self.file.url)
+        elif file:
+            url = self.file.download()
+
         return Para([Image(('', [], []), caption.to_pandoc(), (url, ''))])
 
 
 class File():
     def __init__(self, obj):
-        if obj['type'] == "file":
-            self.type = "file"
-            self.url = obj['file']['url']
-            self.expiry_time = obj['file']['expiry_time']
-        elif obj['type'] == "external":
-            self.type = "external"
-            self.url = obj['external']['url']
+        for type in ["file", "external"]:
+            if obj['type'] == type:
+                self.type = type
+                self.url = obj[type]['url']
+                self.expiry_time = obj[type]['expiry_time'] if type == "file" else None
```

Comment:
I'm not sure that you need the for loop and the if statement. This might me a better way to do the same thing.

```python
self.type = obj['type']
self.url = obj[self.type]['url']
self.expiry_time = obj[self.type]['expiry_time'] if self.type == "file" else None
```